.githooks/pre-commit

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+#set -x
+PREFIX="pre-commit:"
+RESULTS_FILE=tfsec-scan-output.txt
+TFSEC_RESULTS_DIR=/tmp/tfsec-results
+TFSEC_WORKING_DIR=/tmp/tfsec-files
+
+initialise() {
+    mkdir -p $TFSEC_WORKING_DIR
+    mkdir -p $TFSEC_RESULTS_DIR
+    rm  $TFSEC_WORKING_DIR/* 2> /dev/null
+    rm  $TFSEC_RESULTS_DIR/* 2> /dev/null
+    ALLOW_TFSEC_ISSUES="no"
+}
+
+populateGlobalVars() {
+    stagedFileList=$(git diff --diff-filter=d --cached --name-only)
+    tfFileList=$(echo "$stagedFileList" | grep -E '\.(tf)$')
+    if [[ ! -z "$tfFileList" ]] && [ ${#tfFileList[@]} -gt 0 ]; 
+    then
+        TERRAFORM_FILES_STAGED="yes"
+    else
+        TERRAFORM_FILES_STAGED="no"
+    fi
+
+}
+
+copyTerraformFilesToBeScanned() {
+
+    echo "$PREFIX Performing a tfsec scan on the terraform files you have staged (only terraform .tf files will be parsed)."
+    echo "$PREFIX The list of files to be scanned by tfsec will be as follows.."
+
+    #browse through ALL files , not only the TF files
+    for value in ${stagedFileList}
+    do
+        # copy all TF files to /tmp/tfsec-files
+        if [[ $value == *.tf ]]
+        then
+	        echo "$PREFIX FILE: $value"
+            cp "$value" "$TFSEC_WORKING_DIR"
+        fi
+        
+        # check if the tfsec-scan-output file has been staged
+        if [[ $value == $RESULTS_FILE ]]
+        then
+            # by staging this file, the user has indicated that they want to allow through the commit despite the tfsec issues.
+            ALLOW_TFSEC_ISSUES="yes"
+            # also copy this to the same directory so that a diff can take place later between this staged file and the latest scan output
+            cp "$value" "$TFSEC_WORKING_DIR"
+            
+        fi
+    done
+}
+
+removeTimingStatsFromScanOutput() {
+    #remove any timing stats from the file, as these are problematic when doing a diff
+    #Firstly, save the file before any sed replacement just in case we wish to troubleshoot
+    cp $TFSEC_RESULTS_DIR/$RESULTS_FILE $TFSEC_RESULTS_DIR/tfsec-scan-output-orig.txt 
+    sed -i '/^  disk i/d' $TFSEC_RESULTS_DIR/$RESULTS_FILE
+    sed -i '/^  parsing/d' $TFSEC_RESULTS_DIR/$RESULTS_FILE
+    sed -i '/^  adaptation/d' $TFSEC_RESULTS_DIR/$RESULTS_FILE
+    sed -i '/^  checks/d' $TFSEC_RESULTS_DIR/$RESULTS_FILE
+    sed -i '/^  total/d' $TFSEC_RESULTS_DIR/$RESULTS_FILE
+}
+
```

Comment:
Two blank lines here, but elsewhere you use a single blank line between code blocks.

Reply:
ok